sechub-doc/src/main/java/com/daimler/sechub/docgen/usecase/PDSUseCaseAsciiDocGenerator.java

```diff
@@ -0,0 +1,307 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.daimler.sechub.docgen.GeneratorConstants;
+import com.daimler.sechub.docgen.messaging.UseCaseEventOverviewPlantUmlGenerator;
+import com.daimler.sechub.docgen.usecase.UseCaseModel.UseCaseDefGroup;
+import com.daimler.sechub.docgen.usecase.UseCaseModel.UseCaseEntry;
+import com.daimler.sechub.docgen.usecase.UseCaseModel.UseCaseEntry.UseCaseEntryStep;
+import com.daimler.sechub.docgen.usecase.UseCaseModel.UseCaseModelType;
+import com.daimler.sechub.pds.usecase.PDSUseCaseGroup;
+import com.daimler.sechub.sharedkernel.Step;
+import com.daimler.sechub.sharedkernel.usecases.UseCaseGroup;
+
+public class PDSUseCaseAsciiDocGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PDSUseCaseAsciiDocGenerator.class);
+    private static final Pattern PATTERN_MATCH_UNDERSCORE = Pattern.compile("_");
+    // [options="header",cols="1,1,1"]
+    // |===
+    // |HeadA |HeadB |HeadC
+    // //----------------------
+    // |Row1A |Row1B |Row1C
+    // |Row2A |Row2B |Row2Cs
+    // |Row3A |Row3B |Row3C
+    // |===
+    private boolean connectUsecaseWithMessage;
+    private boolean showEventDiagrams;
+
+    public String generateAsciidoc(UseCaseModel model, File diagramsGenFolder) {
+        return generateAsciidoc(model, diagramsGenFolder, true, true);
+    }
+
+    public String generateAsciidoc(UseCaseModel model, File diagramsGenFolder, boolean connectUsecaseWithMessage, boolean showEventDiagrams) {
+        this.connectUsecaseWithMessage = connectUsecaseWithMessage;
+        this.showEventDiagrams = showEventDiagrams;
+
+        Context context = new Context();
+        context.diagramsGenFolder = diagramsGenFolder;
+        int h = 3;
+        context.addLine("");
+        context.addLine(":usecasedoc:");
+        context.addLine("");
+        context.addLine("[[section-usecases]]");
+        context.addLine(headline(h++) + "Use cases");
+        context.addLine("");
+        context.addLine("ifdef::techdoc[]");
+        context.addLine("TIP: The complete documentation about use cases is generated. If you want to change content, please search for ");
+        context.addLine("     `@UseCaseDefinition` references in source code and make necessary changes inside code!");
+        context.addLine("endif::techdoc[]");
+        context.addLine("");
+        generateOverview(model, context, h);
+        generateDetails(model, context, h);
+
+        return context.getAsciiDoc();
+    }
+
+    private void generateDetails(UseCaseModel model, Context context, int h) {
+        for (UseCaseEntry entry : model.getUseCases()) {
+            generateUseCase(context, h, entry);
+        }
+    }
+
+    private void generateOverview(UseCaseModel model, Context context, int h) {
```

Comment:
If have explained that before.
I have changed h to headlineLevel now, even when I am not happy to

``` 
context.addLine(headline(headlineLevel + 1) + group.title);
```
instead of former easier to read:
``` 
context.addLine(headline(h + 1) + group.title);
```